### ISSUE TITLE
Does no work for me

### ISSUE THREAD

**Issue by Esokrates** [TelegramTools/TLMerger#1]
When running your script I obtain the following:
```
Something went wrong in Telegram's side. This is the full exception:

__init__() got an unexpected keyword argument 'max_value'
This part of the process can't be recovered. You must start from scratch.

You can report this issue at https://github.com/TelegramTools/TLMerger/issues/new. Please, give as much details as possible of the error message and attach the 'TLMerger-log.log' file, as all the detailed information about the bug has been written there.

Press ENTER to close the app...Traceback (most recent call last):
  File "TLMerger.py", line 1021, in GatherAllMessages
    bar = progressbar.ProgressBar(max_value=count)
TypeError: __init__() got an unexpected keyword argument 'max_value'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "TLMerger.py", line 3401, in <module>
    GatherAllMessages(ChosenChat)
  File "TLMerger.py", line 1221, in GatherAllMessages
    getpass.getpass("This part of the process can't be recovered. You must start from scratch.\n\nYou can report this issue at https://github.com/TelegramTools/TLMerger/issues/new. Please, give as much details as possible of the error message and attach the 'TLMerger-log.log' file, as all the detailed information about the bug has been written there.\n\nPress ENTER to close the app...")
  File "/usr/lib/python3.7/getpass.py", line 77, in unix_getpass
    passwd = _raw_input(prompt, stream, input=input)
  File "/usr/lib/python3.7/getpass.py", line 146, in _raw_input
    line = input.readline()
KeyboardInterrupt
```

**Comment by ferferga** [comment]
Are you running the Windows .exe file or running the script with Python in your environment?

**Comment by Esokrates** [comment]
@ferferga Running on Ubuntu 19.04, running the .py file with python3.

**Comment by ferferga** [comment]
Seems you didn't run ``pip3 install -r requirements.txt``. Or you did, but you had another version of progressBar2 installed. Run ``pip3 uninstall progressBar2`` and then re-run ``pip3 install -r requirements.txt``

**Comment by Esokrates** [comment]
@ferferga Ok progressbar was already preinstalled, so I followed your instructions, now getting a different problem:
```
All the messages were retrieved. Backing up data into the database...

 47% (304 of 637) |##########            | Elapsed Time: 0:00:01 ETA:   0:00:01Something went wrong in Telegram's side. This is the full exception:

FILEREF_UPGRADE_NEEDED
Press ENTER to try again: 
```

**Comment by ferferga** [comment]
@Esokrates Show me Telethon version using pip freeze

**Comment by Esokrates** [comment]
```
apparmor==2.13.2
bleach==3.1.0
certifi==2018.8.24
cffi==1.13.2
chardet==3.0.4
cryptography==2.8
cupshelpers==1.0
cycler==0.10.0
decorator==4.3.0
defer==1.0.6
defusedxml==0.5.0
devscripts==2.19.4
distro-info===0.21ubuntu2
entrypoints==0.3
evdev==1.1.2
gpg==1.12.0
html5lib==1.0.1
httplib2==0.11.3
idna==2.6
iotop==0.6
ipykernel==4.9.0
ipython==5.8.0
ipython-genutils==0.2.0
jellyfish==0.5.6
Jinja2==2.10
jsonschema==2.6.0
jupyter-client==5.2.3
jupyter-console==5.2.0
jupyter-core==4.4.0
kiwisolver==1.0.1
language-selector==0.1
LibAppArmor==2.13.2
macaroonbakery==1.2.1
Mako==1.0.7
MarkupSafe==1.1.0
matplotlib==3.0.2
meld==3.20.0
mistune==0.8.4
mpmath==1.1.0
munkres==1.1.1
musicbrainzngs==0.6
mutagen==1.40.0
nbconvert==5.4.0
nbformat==4.4.0
netifaces==0.10.4
notebook==5.7.4
numpy==1.16.2
pandocfilters==1.4.2
pexpect==4.6.0
pickleshare==0.7.5
Pillow==5.4.1
progressbar2==3.39.2
prometheus-client==0.5.0
prompt-toolkit==1.0.15
protobuf==3.6.1
pyaes==1.6.1
pyAesCrypt==0.4.2
pyasn1==0.4.8
pycairo==1.16.2
pycparser==2.19
pycups==1.9.73
pycurl==7.43.0.2
Pygments==2.3.1
PyGObject==3.32.0
pymacaroons==0.13.0
PyNaCl==1.3.0
pyparsing==2.2.0
PyQt5==5.12.1
pyRFC3339==1.1
PySimpleSOAP==1.16.2
PySocks==1.6.8
python-apt==1.8.4
python-dateutil==2.7.3
python-debian==0.1.34
python-debianbts==2.8.2
python-utils==2.3.0
pytz==2018.9
pyudev==0.21.0
PyYAML==3.13
pyzmq==17.1.2
reportbug===7.5.2ubuntu1
reportlab==3.5.18
requests==2.21.0
requests-unixsocket==0.1.5
rsa==4.0
scipy==1.2.1
Send2Trash==1.5.0
simplegeneric==0.8.1
six==1.12.0
sympy==1.3
Telethon==0.19.1.6
terminado==0.8.1
testpath==0.4.2
torbrowser-launcher==0.3.1
tornado==5.1.1
traitlets==4.3.2
ubuntu-advantage-tools==19.2
ubuntu-drivers-common==0.0.0
ufw==0.36
Unidecode==1.0.23
urllib3==1.24.1
wcwidth==0.1.7
webencodings==0.5.1
xkit==0.0.0
youtube-dl==2019.1.17
```

**Comment by ferferga** [comment]
@Esokrates Run pip3 install -U telethon and edit the script changing telethon to telethon.sync in line 21.

**Comment by Esokrates** [comment]
@ferferga 
```
Traceback (most recent call last):
  File "TLMerger.py", line 72, in <module>
    client1 = TelegramClient('User1', api_id, api_hash, device_model=TLdevice_model, system_version=TLsystem_version, app_version=TLapp_version, lang_code=TLlang_code, system_lang_code=TLsystem_lang_code, spawn_read_thread=False, update_workers=1)
TypeError: __init__() got an unexpected keyword argument 'spawn_read_thread'
```

**Comment by ferferga** [comment]
@Esokrates Forgot that you also have to remove the last two arguments from line 72.

**Comment by Esokrates** [comment]
@ferferga 
```
Creating database...
Created database successfully!
Traceback (most recent call last):
  File "TLMerger.py", line 3398, in <module>
    CreateTables(database)
  File "TLMerger.py", line 954, in CreateTables
    date = str(datetime.datetime.today())
AttributeError: type object 'datetime.datetime' has no attribute 'datetime'
```

**Comment by Esokrates** [comment]
@ferferga Ok fixed the call in line 954, but now I'm confused:
```
All the messages were retrieved. Backing up data into the database...

100% (637 of 637) |#########################################################################################################################################################| Elapsed Time: 0:00:00 Time:  0:00:00

Done! Backed up  637  messages and  37  files.


Checking the database's data...
Nothing wrong detected. You are good to go!
Do you want to import a conversation using two users, in a 1:1 format? Otherwise, the chat will be imported inside your 'Saved Messages'.

WARNING: Importing into Saved Messages hasn't been tested extensively, but should work. Use at your own risk.
Enter your answer [y/n]: 
```
I do not understand the question.

EDIT: To be more precise: When do I select the target chat it should be merged into? The only thing I was asked was to specifiy a name as the chat I want to merged into another was a Deleted Account because the contact changed number and now I want to merge the old history into the new number. When I enter y will it ask me for the target or will it start right away to do something?

**Comment by ferferga** [comment]
@Esokrates Read https://github.com/TelegramTools/TLMerger/blob/master/README.md#how-does-it-work

You can also take a look at https://github.com/TelegramTools/TLImporter/blob/python/README.md and TLImporter's wiki, where there are examples of how TLImporter and TLMerger works (they are the same concept, but TLImporter uses a txt as the source while TLMerger uses another Telegram chat as the source)

**Comment by Esokrates** [comment]
@ferferga Tried it and got
```
All the messages were retrieved. Backing up data into the database...

100% (637 of 637) |####################################################| Elapsed Time: 0:00:00 Time:  0:00:00

Done! Backed up  637  messages and  37  files.


Checking the database's data...
Nothing wrong detected. You are good to go!
Do you want to import a conversation using two users, in a 1:1 format? Otherwise, the chat will be imported inside your 'Saved Messages'.

WARNING: Importing into Saved Messages hasn't been tested extensively, but should work. Use at your own risk.
Enter your answer [y/n]: y

You have chosen to merge the conversation using two accounts

Now, you have to log in your partner in Telegram...

Do you want to log in your partner using the 'Secret Mode'? Refer to the documentation in GitHub (check https://github.com/TelegramTools/TLSecret as well) for more information. [y/n]: n


Your partner is logged in as <ME,CENSORED>


You are going to copy a conversation from <ME,CENSORED> to <ME,CENSORED> Now, before merging the chat, you need to set up a few settings:


Here are your current settings:

> 1. Timestamp Mode: Default
  > Format: YYYY/MM/DD
  > Seconds: Seconds will be added to timestamps, like this: HH:MM:SS
  > Position: The timestamp will be added to the beginning of each message
> 2. Add hashtags to each message: No
> 3. Attach all Link Previews: No
> 4. Backup of the database in 'Saved Messages': Yes
> 5. Delete original messages: No
> 6. Add 'Forwarded from' header to messages: Yes

> Available commands: 
  !C: Confirm these settings and start the merging of the chats
  !1: See a description and change timestamps settings
  !2: See a description and change hashtags settings
  !3: See a description and change Link Previews settings
  !4: See a description and change Database settings
  !5: See a description and switch between deleting messages or not
  !6: See a description and switch between adding the 'Forwarded from' header to messages or not
Enter a command: C

The command you entered was not valid. Please, enter a valid one: !C

Take in mind that this might trigger some flood limits in Telegram. Press ENTER to continue: 

Also, please, don't interact or chat with your partner while TLMerger is copying messages. This can lead to problems.
Press ENTER to continue: 
STARTED! Now copying the conversation in Telegram...


Processing...

We need to get some data from Telegram before starting. Preparing, this might take a while...
N/A% (0 of 1) |                                                        | Elapsed Time: 0:00:00 ETA:  --:--:--Traceback (most recent call last):
  File "TLMerger.py", line 3541, in <module>
    ExportMessages()
  File "TLMerger.py", line 1289, in ExportMessages
    request = SendMediaRequest(user2, media=InputMediaPhoto(id=InputPhoto(id=photoid, access_hash=PhotoAccHash[index])), message="")
TypeError: __init__() missing 1 required positional argument: 'file_reference'
100% (1 of 1) |########################################################| Elapsed Time: 0:00:00 Time:  0:00:00

```

I could not figure out where to select the target chat I am going to merge into, it just wanted to start copying. I expected  a dialog to ask me for the number of the target like for the source, but that did not happen.

**Comment by Esokrates** [comment]
@ferferga Your documentation states

> · Choose the chat where do you want to copy the messages (the destination dialog)

When do I get to choose this? See my log, it never asked me.

**Comment by ferferga** [comment]
@Esokrates That's only for TLImporter. TLMerger copies to Saved Messages or to your chat with another account, so you can't log in yourself. Im not sure what are you trying achieve, and I'm starting to think that TLMerger is not exactly what do you want.

Seems as well that Telegram did some API changes that TLMerger can't handle.

**Comment by ferferga** [comment]
@Esokrates Also please, if possible, try to run a VM with Windows and run the executable there. You don't have the same Python version that I used to build (that triggered the datetime error) and running that exe it's the best way to ensure that you have the right environment.

**Comment by Esokrates** [comment]
@ferferga What I'm trying to achieve: I have a contact that changed his phone number 1 year ago and created a new telegram account with his new phone number. Now I want to merge the old chat (Deleted Account in the meantime) with the new chat. This is exactly what your README gives as an example.

So please let me asked, when is the the tool supposed to ask me to specify the target chat? I do not want login my partner, I just want it to copy stuff to the new chat with the new number in MY account.

**Comment by ferferga** [comment]
@Esokrates You simply can't do that. Telegram doesn't allow you to do this. That's why you need to overcome this limitation by logging in your partner. TLMerger will then reconstruct the chat there.

**Comment by Esokrates** [comment]
@ferferga That's unfortunate, could you state that more clearly in the README?

**Comment by ferferga** [comment]
Sure. You might still want to take a look at Secret Mode, it should provide a hassle-free experience while authenticating the other user.
